packages/flutter/lib/src/cupertino/radio.dart

```diff
@@ -235,47 +194,88 @@ class CupertinoRadio<T> extends StatefulWidget {
   /// {@macro flutter.widgets.Focus.autofocus}
   final bool autofocus;
 
-  bool get _selected => value == groupValue;
-
   @override
   State<CupertinoRadio<T>> createState() => _CupertinoRadioState<T>();
 }
 
-class _CupertinoRadioState<T> extends State<CupertinoRadio<T>>
-    with TickerProviderStateMixin, ToggleableStateMixin {
-  final _RadioPainter _painter = _RadioPainter();
-
-  bool focused = false;
-
-  void _handleChanged(bool? selected) {
-    if (selected == null) {
-      widget.onChanged!(null);
-      return;
-    }
-    if (selected) {
-      widget.onChanged!(widget.value);
-    }
-  }
+class _CupertinoRadioState<T> extends State<CupertinoRadio<T>> {
+  FocusNode get _effectiveFocusNode => widget.focusNode ?? (_internalFocusNode ??= FocusNode());
+  FocusNode? _internalFocusNode;
 
   @override
   void dispose() {
-    _painter.dispose();
+    _internalFocusNode?.dispose();
     super.dispose();
   }
 
   @override
-  ValueChanged<bool?>? get onChanged => widget.onChanged != null ? _handleChanged : null;
+  Widget build(BuildContext context) {
+    final WidgetStateProperty<MouseCursor> effectiveMouseCursor =
+        WidgetStateProperty.resolveWith<MouseCursor>((Set<WidgetState> states) {
+          return WidgetStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states) ??
+              (states.contains(WidgetState.disabled)
+                  ? SystemMouseCursors.basic
+                  : kIsWeb
+                  ? SystemMouseCursors.click
+                  : SystemMouseCursors.basic);
```

Comment:
Nit: Would it be any more clear if this was a switch statement?

Reply:
Would this be cleaner if I have it right?
```dart
!states.contains(WidgetState.disabled) && kIsWeb
  ? SystemMouseCursors.click
  : SystemMouseCursors.basic
```